packages/middleware-stack/src/MiddlewareStack.ts

```diff
@@ -236,17 +239,31 @@ export const constructStack = <Input extends object, Output extends object>(): M
     ): MiddlewareStack<InputType, OutputType> => {
       const cloned = cloneTo(constructStack<InputType, OutputType>());
       cloned.use(from);
+      cloned.identifyOnResolve(identifyOnResolve || cloned.identifyOnResolve() || from.identifyOnResolve());
       return cloned;
     },
 
     applyToStack: cloneTo,
 
     identify: (): string[] => {
       return getMiddlewareList(true).map((mw: MiddlewareEntry<Input, Output>) => {
-        return mw.name + ": " + (mw.tags || []).join(",");
+        const step =
+          mw.step ??
+          ((mw as unknown) as RelativeMiddlewareOptions).relation +
+            " " +
+            ((mw as unknown) as RelativeMiddlewareOptions).toMiddleware;
+        return mw.name + " - " + step;
       });
     },
 
+    identifyOnResolve(toggle?: boolean) {
+      if (typeof toggle !== "boolean") {
+        return identifyOnResolve;
+      }
+      identifyOnResolve = !!toggle;
+      return identifyOnResolve;
+    },
```

Comment:
nit: after the `typeof toggle` boolean check, does `toggle` need to have `!!`?

```suggestion
    identifyOnResolve(toggle?: boolean) {
      if (typeof toggle === "boolean")
        identifyOnResolve = toggle;
      return identifyOnResolve;
    },
```